### PR TITLE
Feature/mappers

### PR DESCRIPTION
**Pull request by ortiztena**
* added manfred model
* added mappers from manfred to europass
* added test to mappers

### CODE REVIEW

File: src/mappers/manfred-to-europass.mapper.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            },
+            workExperienceList: [], // external function
+            skills: [], // external function
+        },
+    }
+    return europass;
+}
+
+
+export const mapFromModelToEuropass = (cv: ManfredAwesomicCV) =>
```
Comment (brauliodiez) [inline src/mappers/manfred-to-europass.mapper.ts]: Rename this variable to 
manfredCv

File: src/mappers/manfred-to-europass.mapper.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    cv ? manfredToEuropass(cv) : {};
```
Comment (brauliodiez) [inline src/mappers/manfred-to-europass.mapper.ts]: Instead of returning empty object return null

File: src/mappers/manfred-to-europass.mapper.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export const manfredToEuropass = (manfred: ManfredAwesomicCV): SkillsPassport => {
+    const europass: SkillsPassport = {
+        documentInfo: {
+            documentType: 'ECV',
+            creationDate: new Date(Date.now()),
+            lastUpdateDate: new Date(Date.now()),
+            xsdVersion: 'V3.0',
+            generator: 'EWA',
+            comment: 'Europass CV'
+        },
```
Comment (brauliodiez) [inline src/mappers/manfred-to-europass.mapper.ts]: Crate a separate function for this, this should be:

```typescript
    const europass: SkillsPassport = {
        documentInfo: generateEuropassDocumentInfo(),
```

```typescript
const generateEuropassDocumentInfo = () :DocumentInfo => {
  return {
            documentType: 'ECV',
            creationDate: new Date(Date.now()),
            lastUpdateDate: new Date(Date.now()),
            xsdVersion: 'V3.0',
            generator: 'EWA',
            comment: 'Europass CV'
  }
}
```

File: src/mappers/manfred-to-europass.mapper.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        printingPreferences: {
+            type: ''
+        },
```
Comment (brauliodiez) [inline src/mappers/manfred-to-europass.mapper.ts]: Same here

File: src/mappers/manfred-to-europass.mapper.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        learnerInfo: {
+            identification:
+            {
```
Comment (brauliodiez) [inline src/mappers/manfred-to-europass.mapper.ts]: Same for identification but passing the function manfred.aboutMe

File: src/models/europass/index.ts [hunk 6]
```diff
@@ -5,9 +5,9 @@ import { generateDocumentInfoElement } from "@/engine/document-info.engine";
 
 // Root
 export interface SkillsPassport {
-  documentInfo: DocumentInfo;
-  printingPreferences: PrintingPreferences;
-  learnerInfo: LearnerInfo;
+  documentInfo?: DocumentInfo;
+  printingPreferences?: PrintingPreferences;
+  learnerInfo?: LearnerInfo;
```
Comment (brauliodiez) [inline src/models/europass/index.ts]: Why is this optional on Skills Passport?

File: src/models/europass/learnerInfo/skills.model.ts [hunk 7]
```diff
@@ -3,7 +3,7 @@ export interface Skills {
 }
 
 interface Linguistic {
-  motherTonguiList: MotherTongue[];
+  motherTongList: MotherTongue[];
```
Comment (brauliodiez) [inline src/models/europass/learnerInfo/skills.model.ts]: Typo

motherTongueList

File: src/mappers/manfred-to-europass.mapper.ts (hunk 1 above)
Comment (ortiztena) [reply src/mappers/manfred-to-europass.mapper.ts]: done

File: src/mappers/manfred-to-europass.mapper.ts (hunk 2 above)
Comment (ortiztena) [reply src/mappers/manfred-to-europass.mapper.ts]: done

File: src/mappers/manfred-to-europass.mapper.ts (hunk 3 above)
Comment (ortiztena) [reply src/mappers/manfred-to-europass.mapper.ts]: done

File: src/mappers/manfred-to-europass.mapper.ts (hunk 4 above)
Comment (ortiztena) [reply src/mappers/manfred-to-europass.mapper.ts]: done

File: src/models/europass/learnerInfo/skills.model.ts (hunk 7 above)
Comment (ortiztena) [reply src/models/europass/learnerInfo/skills.model.ts]: fixed

File: src/models/europass/index.ts (hunk 6 above)
Comment (ortiztena) [reply src/models/europass/index.ts]: turned it back to mandatory

File: src/mappers/manfred-to-europass.mapper.ts (hunk 5 above)
Comment (ortiztena) [reply src/mappers/manfred-to-europass.mapper.ts]: nested  learnedInfo & identification.
created generate personName function.
should i keep nesting others properties like contactInfo & demographics ?

File: src/mappers/manfred-to-europass.mapper.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        email: {
+            contact: "",
+        },
+    },
+    demographics: {
+        birthDate: new Date(cv.aboutMe.profile.birthday),
+    },
+});
+
+const generatePersonName = (profile): Identification["personName"] => ({
```
Comment (brauliodiez) [inline src/mappers/manfred-to-europass.mapper.ts]: It should be

```diff
- const generatePersonName = (profile): Identification["personName"] => ({
+ const generatePersonName = (profile): PersonName => ({
```

You need to export _PersonName_ interface in _identification.model.ts_

```diff
- interface PersonName {
+ interface PersonName {

  firstName: string;
  surName: string;
}
```

Comment (ortiztena) [reply src/mappers/manfred-to-europass.mapper.ts]: Done.
Yes, way better, thanks.

**Pull request merged**